### PR TITLE
Bump go-storage to 4.3.2

### PR DESCRIPTION
**Pull request by JinnyYi**

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -86,6 +88,12 @@ func formatError(err error) error {
 		return err
 	}
 
+	if pe, ok := err.(*fs.PathError); ok {
```
Comment (Xuanwo) [inline utils.go:91]: go 1.15 doesn't have `io/fs` support, please use `os.PathError` instead.

File: utils.go [hunk 2]
```diff
@@ -86,6 +88,12 @@ func formatError(err error) error {
...
+		if errno, ok := pe.Err.(syscall.Errno); ok && errno == syscall.EISDIR {
```
Comment (Xuanwo) [inline utils.go:92]: `syscall` may not work on `darwin` and `windows`.

Comment (JinnyYi) [reply utils.go:91]: I'm not sure how to handle `PathError` here. Can I use the `is a directory` error message as the condition?

File: storage.go [hunk 3]
```diff
@@ -64,6 +65,14 @@ func (s *Storage) copy(ctx context.Context, src string, dst string, opt pairStor
 
 	dstFile, needClose, err := s.createFile(rd)
 	if err != nil {
+		if pe, ok := err.(*os.PathError); ok {
```
Comment (Xuanwo) [inline storage.go:68]: It's not a trivial change, we need a new PR to solve this problem.

File: storage.go [hunk 4]
```diff
@@ -195,6 +205,12 @@ func (s *Storage) move(ctx context.Context, src string, dst string, opt pairStor
 
 	err = os.Rename(rs, rd)
 	if err != nil {
+		if errors.Is(err, os.ErrExist) {
```
Comment (JinnyYi) [inline storage.go:208]: I tested on Windows 10 and got the syscall.Errno `ERROR_ACCESS_DENIED (5)`. Actually the file and dir can be accessed. I tried to call `os.Chmod` after `os.MkdirAll` and `os.Create`, but it didn't work.

File: utils.go (hunk 2 above)
Comment (Xuanwo) [reply utils.go]: I don't think use `is a directory` is a good idea here.

How about:

```go
	if pe, ok := err.(*os.PathError); ok {
		switch pe.Err {
		case syscall.EISDIR:
			return fmt.Errorf("%w: %v", services.ErrObjectModeInvalid, err)
		}
	}
```

**Comment by Xuanwo** [comment]
I found it impossible to handle the error in `formatError`, so how about checking them before we call `os.Rename`?

Please take https://github.com/beyondstorage/go-service-fs/tree/test-windows a look.

**Comment by Xuanwo** [comment]
By the way, this fix is outside the scope of `Bump go-storage to v4.3.2`. How about split them into two PRs:

- Bump go-storage
- Fix copy and move behavior(in this PR we bump integration-tests)

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**